### PR TITLE
[IR] Fix for https://youtrack.jetbrains.com/issue/KT-44219 + test

### PR DESCRIPTION
**Pull request by homuroll**

### CODE REVIEW

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/PostInlineLowering.kt [hunk 1]
```diff
@@ -5,35 +5,41 @@
...
+internal class PostInlineLowering(val context: Context) : BodyLoweringPass {
 
     private val symbols get() = context.ir.symbols
 
-    override fun lower(irFile: IrFile) {
-        irFile.transformChildrenVoid(object : IrBuildingTransformer(context) {
+    override fun lower(irBody: IrBody, container: IrDeclaration) {
+        val irFile = container.file
+        val irBuilder = context.createIrBuilder((container as IrSymbolOwner).symbol)
+        irBody.transformChildrenVoid(object : IrElementTransformerVoid() {
```
Comment (SvyatoslavScherbina) [inline backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/PostInlineLowering.kt:37]: Would it traverse local function bodies multiple times?

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/PostInlineLowering.kt [hunk 2]
```diff
@@ -5,35 +5,41 @@
...
+import org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid
 import org.jetbrains.kotlin.ir.visitors.transformChildrenVoid
 
 /**
  * This pass runs after inlining and performs the following additional transformations over some operations:
  *     - Convert immutableBlobOf() arguments to special IrConst.
  *     - Convert `obj::class` and `Class::class` to calls.
  */
-internal class PostInlineLowering(val context: Context) : FileLoweringPass {
+internal class PostInlineLowering(val context: Context) : BodyLoweringPass {
```
Comment (SvyatoslavScherbina) [inline backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/PostInlineLowering.kt:30]: Why is `BodyLoweringPass` required here?

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/PostInlineLowering.kt (hunk 1 above)
Comment (homuroll) [reply backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/PostInlineLowering.kt:37]: No, by default `BodyLoweringPass` does not go into the bodies. (see https://github.com/JetBrains/kotlin/blob/master/compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/Lower.kt#L121)

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/PostInlineLowering.kt (hunk 2 above)
Comment (homuroll) [reply backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/PostInlineLowering.kt:30]: We want to know the containing function.

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/PostInlineLowering.kt (hunk 1 above)
Comment (SvyatoslavScherbina) [reply backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/PostInlineLowering.kt:37]: I see. But isn't `irBuilder`'s scope then wrong for local functions?

Comment (homuroll) [reply backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/PostInlineLowering.kt:37]: Indeed, we have to the same trick as in `PreInlineLowering`.

**Review by SvyatoslavScherbina — APPROVED** [review]

**Pull request merged**